File: app/src/main/java/com/beekay/thoughts/adapter/RemindersAdapter.java
package com.beekay.thoughts.adapter;

import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.beekay.thoughts.R;
import com.beekay.thoughts.model.Reminder;

import java.util.List;

public class RemindersAdapter extends RecyclerView.Adapter<RemindersAdapter.RemindersViewHolder> {

    private List<Reminder> reminders;

    public RemindersAdapter(List<Reminder> reminders) {
        this.reminders = reminders;
    }

    @NonNull
    @Override
    public RemindersViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.reminder_item,parent,false);
        return new RemindersAdapter.RemindersViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RemindersViewHolder holder, int position) {
        Reminder reminder = reminders.get(position);
        if (reminder.isStatus()) {
            holder.constraintLayout.setBackgroundColor(Color.parseColor("#AACCAA"));
        } else {
            holder.constraintLayout.setBackgroundColor(Color.parseColor("#FFAAAA"));
        }
        holder.reminderField.setText(reminder.getReminderText());
        holder.statusField.setText("Status : " + (reminder.isStatus() ? "Completed" : "Incomplete"));
        holder.toBeDoneOnField.setText("To Be Done On \n" + reminder.getToBeDoneOn());
        holder.createdOnField.setText("Created On \n" + reminder.getCreatedOn());
    }

    @Override
    public int getItemCount() {
        return reminders.size();
    }

    class RemindersViewHolder extends RecyclerView.ViewHolder {

        TextView reminderField;
        TextView toBeDoneOnField;
        TextView createdOnField;
        TextView statusField;
        ConstraintLayout constraintLayout;

        public RemindersViewHolder(View itemView) {
            super(itemView);
            reminderField = itemView.findViewById(R.id.reminder_text);
            statusField = itemView.findViewById(R.id.reminder_status);
            toBeDoneOnField = itemView.findViewById(R.id.scheduled_for);
            createdOnField = itemView.findViewById(R.id.createdOn);
            constraintLayout = itemView.findViewById(R.id.rConstraintLayout);
        }
    }
}

File: app/src/main/java/com/beekay/thoughts/adapter/RemindersAdapter.kt
package com.beekay.thoughts.adapter

import android.annotation.SuppressLint
import android.graphics.Color
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.beekay.thoughts.databinding.ItemReminderBinding
import com.beekay.thoughts.model.Reminder
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/**
 * Created by Krishna by 21-11-2020
 */
class RemindersAdapter : RecyclerView.Adapter<RemindersAdapter.ReminderViewHolder>() {

    private val reminders: MutableList<Reminder> = mutableListOf()
    private val currentDate = LocalDateTime.now()

    @SuppressLint("NotifyDataSetChanged")
    fun setReminders(reminders: List<Reminder>) {
        this.reminders.clear()
        this.reminders.addAll(reminders)
        notifyDataSetChanged()
    }

    class ReminderViewHolder(
        private val binding: ItemReminderBinding,
        private val currentDate: LocalDateTime
    ) : RecyclerView.ViewHolder(binding.root) {
        private val dateFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm")

        fun bind(reminder: Reminder) {
            binding.createdOn.text = reminder.createdOn
            binding.scheduledFor.text = reminder.toBeDoneOn
            binding.reminderText.text = reminder.reminderAbout
            val remindOn = LocalDateTime.parse(reminder.toBeDoneOn, dateFormatter)
            val status = if (reminder.status) {
                "Done"
            } else {
                if (currentDate.isBefore(remindOn)) {
                    "Pending"
                } else {
                    "Missed"
                }
            }
            binding.reminderStatus.text = status
            if (status == "Done") {
                binding.root.setBackgroundColor(Color.parseColor("#3DA159"))
            } else if (status == "Missed") {
                binding.root.setBackgroundColor(Color.parseColor("#e8416f"))
            }
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReminderViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = ItemReminderBinding.inflate(inflater, parent, false)
        return ReminderViewHolder(binding, currentDate)
    }

    override fun getItemCount() = reminders.size

    override fun onBindViewHolder(holder: ReminderViewHolder, position: Int) {
        holder.bind(reminders[position])
    }
}